source/uri-options/tests/auth-options.json

```diff
@@ -0,0 +1,20 @@
+{
+  "tests": [
+    {
+      "description": "Valid auth options are parsed correctly",
+      "uri": "mongodb://example.com/?authMechanism=GSSAPI&authMechanismProperties=SERVICE_NAME:other,CANONICALIZE_HOST_NAME:true&authSource=$external",
+      "valid": true,
+      "warning": false,
+      "hosts": null,
+      "auth": null,
+      "options": {
+        "authmechanism": "GSSAPI",
+        "authMechanismProperties": {
+          "SERVICE_NAME": "other",
+          "CANONICALIZE_HOST_NAME": "true"
```

Comment:
Why is this string `'true'` instead of boolean `true`?

Reply:
I wasn't sure if we should be prescriptive about how/when drivers parse the authMechanismProperties from strings. Since it's a lot easier to go from the boolean to the string than from the string to the boolean though, I'll update this to use the boolean.